### DISCUSSION TITLE
Invalid workspace tags?

### DISCUSSION THREAD

**Discussion by paulross80** [chalet-org/chalet#255] [Q&A] [answered]
Hello. I'm testing Chalet a liitle bit more. I've found a problem with the project metadata
I have this static library target, Papa, which is my own game engine
I want this target to use the project metadata, because the static library is my base project

```
"Papa": {
			"extends": "papa_code",
			"outputDescription": "Papa library",
			"kind": "staticLibrary",
			"language": "C++",
			"metadata": {
				"name": "${meta:workspaceName}",
				"author": "${meta:workspaceAuthor}",
				"description": "${meta:workspaceDescription}",
				"version": "${meta:workspaceVersion}"
			},
			"files": "cpp/papa_lib/**.cpp",
			"configureFiles": "cpp/papa_lib/config/Papa.hpp.in"
		}
```

But I'm getting this error after running `chalet configure -c Debug`

```
ERROR: Failed to validate the json file: chalet.json
ERROR: An invalid value was found in 'version': '${{meta:workspaceVersion}}'. Expected the pattern '^((\d+\.){{1,3}})?\d+$'
ERROR: An invalid value was found in 'name': '${{meta:workspaceName}}'. Expected the pattern '^[\w\-+ ]+$'
```

Seems to have a problem only with the name and version
These are literally the first three lines of my project file:

```
{
	"name": "Papa",
	"version": "1.0.0.2023",
	"author": "Paul",
```

How can I fix this?

**Comment by rewrking** [comment, marked as answer]
Sorry for the late reply. So basically, you've stumbled upon some accidental / undocumented functionality, but I think the use case is interesting. You basically want to explicitly inherit workspace metadata as the project metadata. so those `PROJCT_*` macros will be the same as the `WORKSPACE_*` counterparts.

Okay, so two things here: 
1. `name` has validation for basic characters and spaces that stop you from using `${meta:workspaceName}`
2. `version` has validation to only expect a semantic version (ex: `1.1.1`)

`version` is a little more tricky, because it's parsed later on into separate macros: - `*_VER_MAJOR`, `*_VER_MINOR`, `*_VER_PATCH`. I just have to disable the validation upfront, and validate the value that resulted from the variable subsitution after `${meta:workspaceVersion}`. I'll have all this fixed in the next version. I'm trying to finish it up this week.

**Reply by rewrking** [reply]
Opened #265 to track the changes that are needed for this to work.

**Comment by paulross80** [comment]
> I'll have all this fixed in the next version

Nice! I'm glad to have helped a bit
